shell/shared/grain-shared.js

```diff
@@ -14,6 +14,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { waitPromise } from "/imports/server/async-helpers.js";
```

Comment:
Doesn't this fail on the client? This file runs on both server and client (for client-side prediction purposes). I think you need to guard your code accordingly.

BTW, meteor-promise adds a `.await()` method to promises which you can use instead of importing `waitPromise()`.